### PR TITLE
enable SDK Analyzer for Azure.Core

### LINKED ISSUE #18083: Re-enable SDK analyzers in Azure.Core
SDK analyzers [are disabled](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Azure.Core.csproj#L12) in Azure.Core. They should be re-enabled and only the required violations should be excluded.

### PR DESCRIPTION
**Pull request by christothes**
closes #18083

### CODE REVIEW

File: sdk/core/Azure.Core/src/GlobalSuppressions.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+[assembly: SuppressMessage("Usage", "AZC0001:Use one of the following pre-approved namespace groups (https://azure.github.io/azure-sdk/registered_namespaces.html): Azure.AI, Azure.Analytics, Azure.Communication, Azure.Data, Azure.DigitalTwins, Azure.Iot, Azure.Learn, Azure.Media, Azure.Management, Azure.Messaging, Azure.Search, Azure.Security, Azure.Storage, Azure.Template, Azure.Identity, Microsoft.Extensions.Azure", Justification = "<Pending>", Scope = "namespace", Target = "~N:Azure.Core.Cryptography")]
+[assembly: SuppressMessage("Usage", "AZC0001:Use one of the following pre-approved namespace groups (https://azure.github.io/azure-sdk/registered_namespaces.html): Azure.AI, Azure.Analytics, Azure.Communication, Azure.Data, Azure.DigitalTwins, Azure.Iot, Azure.Learn, Azure.Media, Azure.Management, Azure.Messaging, Azure.Search, Azure.Security, Azure.Storage, Azure.Template, Azure.Identity, Microsoft.Extensions.Azure", Justification = "<Pending>", Scope = "namespace", Target = "~N:Azure.Core.Shared")]
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/GlobalSuppressions.cs:20]: Is there a way to target the entire assembly?

**Review by pakrym — APPROVED** [review]
Thank you!

File: sdk/core/Azure.Core/src/GlobalSuppressions.cs (hunk 1 above)
Comment (christothes) [reply sdk/core/Azure.Core/src/GlobalSuppressions.cs:20]: Not that I could find.

Comment (pakrym) [reply sdk/core/Azure.Core/src/GlobalSuppressions.cs:20]: What about `Scope = "namespaceanddescendants",`?

Comment (christothes) [reply sdk/core/Azure.Core/src/GlobalSuppressions.cs:20]: On second thought, I was thinking of all namespaces. assembly scope may be possible.

Comment (pakrym) [reply sdk/core/Azure.Core/src/GlobalSuppressions.cs:20]: or `Scope = "module"`

**Pull request merged**